File: source/backend/projects/Pulse.Api/Models/Users/UserDetailsDto.cs
namespace Pulse.Api.Models.Users
{
    using System;
    using Data.Auth;
    using Webfarm.Sdk.Data;

    public class UserDetailsDto : IIdentified<Guid>
    {
        public Guid Id { get; set; }

        public string Email { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string School { get; set; }

        public string Subject { get; set; }

        public string City { get; set; }

        public string State { get; set; }

        public UserRole Role { get; set; }

        public string CreatedBy { get; set; }

        public DateTimeOffset Created { get; set; }

        public string ModifiedBy { get; set; }

        public DateTimeOffset? Modified { get; set; }
    }
}

File: source/frontend/projects/pulse/src/app/api/models/user-details-dto.ts
/* tslint:disable */
/* eslint-disable */
import { UserRole } from './user-role';
export interface UserDetailsDto {
  city?: null | string;
  created?: string;
  createdBy?: null | string;
  email?: null | string;
  firstName?: null | string;
  id?: string;
  lastName?: null | string;
  modified?: null | string;
  modifiedBy?: null | string;
  role?: UserRole;
  school?: null | string;
  state?: null | string;
  subject?: null | string;
}
